src/main/scala/org/broadinstitute/dsde/agora/server/webservice/methods/MethodsService.scala

```diff
@@ -90,9 +91,12 @@ trait MethodsService extends HttpService with PerRequestCreator with AgoraDirect
       post {
         commonNameFromCookie() { commonName =>
           entity(as[AgoraEntity]) { agoraEntity =>
-            requestContext =>
-              val entityWithOwner = agoraEntity.copy(owner = Option(commonName))
-              perRequest(requestContext, methodsAddHandlerProps, ServiceMessages.Add(requestContext, entityWithOwner))
+            val validation = AgoraValidation.validateMetadata(agoraEntity)
+            validate(validation.valid, validation.message) {
```

Comment:
I think we can supply a rejection handler wherein we can wrap the message in AgoraError in order to json serialize. Should be in spray routing docs